### PR TITLE
Feature/86bz4fzn0

### PR DESCRIPTION
**Pull request by ptrodion**

### CODE REVIEW

File: ui/package.json [hunk 1]
```diff
@@ -16,15 +17,24 @@
     "react-dom": "^18.2.0"
   },
   "devDependencies": {
+    "@eslint/compat": "^1.1.0",
     "@types/node": "^20.12.11",
-    "@types/react": "^18.2.66",
-    "@types/react-dom": "^18.2.22",
-    "@typescript-eslint/eslint-plugin": "^7.2.0",
-    "@typescript-eslint/parser": "^7.2.0",
+    "@types/react": "^18.3.3",
```
Comment (anastasia1383) [inline ui/package.json:22]: lock versions pls

**Review by anastasia1383 — CHANGES_REQUESTED** [review]

File: ui/.eslintrc.cjs [hunk 2]
```diff
@@ -1,18 +1,51 @@
...
+        allow: ['error'],
+      },
+    ],
+    'react/react-in-jsx-scope': 'off',
+    'react/display-name': 'off',
+    indent: ['error', 2],
+    'linebreak-style': ['error', 'unix'],
+    quotes: ['error', 'single'],
+    semi: ['error', 'always'],
+    // Правила для определения функциональных компонентов
```
Comment (anastasia1383) [inline ui/.eslintrc.cjs:42]: use english pls

File: ui/package.json [hunk 3]
```diff
@@ -20,20 +21,29 @@
   "devDependencies": {
     "@testing-library/jest-dom": "6.4.6",
     "@testing-library/react": "16.0.0",
-    "@types/node": "20.12.11",
-    "@types/react": "18.2.66",
-    "@types/react-dom": "18.2.22",
-    "@typescript-eslint/eslint-plugin": "7.2.0",
-    "@typescript-eslint/parser": "7.2.0",
-    "@vitejs/plugin-react": "4.2.1",
+    "@eslint/compat": "^1.1.0",
```
Comment (anastasia1383) [inline ui/package.json:24]: lock the versions

File: ui/prettierrc.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "singleQuote": false,
+  "trailingComma": "es5",
+  "jsxBracketSameLine": false,
+  "tabWidth": 2,
+  "semi": true,
+  "printWidth": 80
+}
```
Comment (anastasia1383) [inline ui/prettierrc.json:8]: missing empty line. maybe you can make it a rule? empty line have to be the last for all files

**Review by anastasia1383 — CHANGES_REQUESTED** [review]

File: ui/.eslintrc.cjs (hunk 2 above)
Comment (ptrodion) [reply ui/.eslintrc.cjs:42]: Corrected

File: ui/package.json (hunk 3 above)
Comment (ptrodion) [reply ui/package.json:24]: Corrected

File: ui/.eslintrc.cjs (hunk 2 above)
Comment (ptrodion) [reply ui/.eslintrc.cjs:42]: added rule for last empty line 'eol-last': ['error', 'always'],

Comment (ptrodion) [reply ui/.eslintrc.cjs:42]: and added   "endOfLine": "auto" for prettierrc.json

File: ui/prettierrc.json (hunk 4 above)
Comment (ptrodion) [reply ui/prettierrc.json:8]: The problem is that this is a json file and when saving, the last line is moved to the line with the last character. In the last committee after adding endOfLine": "auto" it is term 9, do I need to add empty line for json exactly ?

**Review by anastasia1383 — APPROVED** [review]

**Review by NatalySheludko — APPROVED** [review]

**Pull request merged**